src/modules/claim/resolvers/credential.resolver.ts

```diff
@@ -104,4 +108,32 @@ export class RoleCredentialResolver implements CredentialResolver {
       })
       .filter(isVerifiableCredential);
   }
+
+  /**
+   * Sets role credential cache
+   * @param roleCredentialcache
+   */
+  setRoleCredentialCache(roleCredentialcache: IRoleCredentialCache): void {
+    this.roleCredentialCache = roleCredentialcache;
+  }
+
+  /**
+   * Fetches DID Document for the given DID
+   * @param did subject DID
+   * @param didDocumentCache Cache to store DIDDocument. Cache is updated with Document retrieved for the DID
+   * @returns
+   */
+  async getDIDDocument(
+    did: string,
+    didDocumentCache?: IDIDDocumentCache
+  ): Promise<IDIDDocument> {
+    const cachedDIDDocument = didDocumentCache?.getDIDDocument(did);
+    if (cachedDIDDocument) {
+      return cachedDIDDocument;
+    }
+    const resolvedDIDDocument =
+      await this.didService.getDIDDocumentFromUniversalResolver(did);
```

Comment:
@nichonien can you let me know if this is implementing the getDIDDoc method correctly?